syntax = "proto3";
option java_multiple_files = true;
option java_outer_classname = "CoreServiceProto";
option objc_class_prefix = "HLW";
package protoStorage;

message Task{
  string name = 1;
  string label = 2;
  enum Priority{
    NONE = 0;
    FIRST = 1;
    SECOND = 2;
    THIRD = 3;
  }
  Priority priority = 3;
  TaskDate date = 4;
}

message TaskEntity{
  Task task = 1;
  bool complete = 2;
  repeated TaskEntity subTasks = 3;
}
message TaskDate{
  uint32 date = 1;
}
message Storage{
  repeated TaskEntity tasks = 1;
}

